### ISSUE TITLE
map.deep-merge() function unsupported?

### ISSUE THREAD

**Issue by Aonrud** [connorskees/grass#76]
Hi,

I'm using grass as part of [Zola](https://github.com/getzola/zola) and it seems it can't compile sass that includes [PicoCSS](https://github.com/picocss/pico).

It looks like the [map.deep-merge()](https://sass-lang.com/documentation/modules/map#deep-merge) function isn't implemented? It's marked 'since 1.27.0' so presume should be in the targeted dart-sass version?

Including Pico 2.0-alpha1 gives this error:

```
Error: Error: Undefined function.
40 │ $breakpoints: map.deep-merge(
   │               ^^^^^^^^^^^^^  
```

**Comment by connorskees** [comment]
Yes, it seems I skipped over implementing [all of the changes added in 1.27.0](https://github.com/sass/dart-sass/releases/tag/1.27.0) and never ended up going back to implement them later -- I'm surprised it's already been 2 years. 

I've pushed up a change that adds this function and I've verified that after this commit it is possible to compile PicoCSS v2. I'll also be adding this library to the corpus I test releases against.

I should be releasing a bugfix version some time today. To get access to this change, however, I assume you'd need to either [build zola from source](https://www.getzola.org/documentation/getting-started/installation/#from-source) (removing the Cargo.lock) or wait for the next zola release.

**Comment by connorskees** [comment]
If you wish to use this function right now, you can build zola from source after updating the `grass` dependency to point to the git repository, `grass = { git = "https://github.com/connorskees/grass" }`

**Comment by Aonrud** [comment]
Thanks! That's an impressively quick response :) Much appreciated.

Just for the sake of completion, and it doesn't affect this case, a quick search of the codebase suggests you might be missing the equivalent [deep-remove()](https://sass-lang.com/documentation/modules/map#deep-remove) function as well.

**Comment by Aonrud** [comment]
Sorry, you already said all of the 1.27.0 changes, so that comment is probably redundant. Anyway, thanks for the response!

**Comment by connorskees** [comment]
This change has been released in version [`0.12.4`](https://github.com/connorskees/grass/releases/tag/v0.12.4). I was hoping to implement all of the 1.27.0 changes, but settled for just adding the new functions `map.deep-merge` and `map.deep-remove`. There's some weird difference between the global and module-scoped functions (e.g. `map-set` vs `map.set`) that I want to investigate before implementing the other changes.
